Dockerfile

```diff
@@ -23,7 +23,11 @@ ARG GO_VERSION="1.13.4"
 #pull dependencies required for downloading and building libndctl
 ARG CACHEBUST
 RUN echo "Updating build image from ${CLEAR_LINUX_BASE} to ${SWUPD_UPDATE_ARG:-the latest release}."
-RUN swupd update ${SWUPD_UPDATE_ARG} && swupd bundle-add ${NDCTL_BUILD_DEPS} c-basic && rm -rf /var/lib/swupd /var/tmp/swupd
+# openssh-client and sudo are needed for CI testing. They are small enough that it doesn't
+# affect developers that they get installed here, but it is a small saving in the CI
+# because then the installation can be cached.
+RUN swupd update ${SWUPD_UPDATE_ARG} && swupd bundle-add ${NDCTL_BUILD_DEPS} c-basic openssh-client sudo && rm -rf /var/lib/swupd /var/tmp/swupd
```

Comment:
@pohly Is not enough installing these packages new packages in `env,.BUILD_IMAGE`? I would prefer if we can avoid ending-up these CI specific packages in the final image.

Reply:
> in the final image

This isn't the final image. It's the build image which gets only used for preparing the binaries which then later get copied into the final image.

This approach is a tradeoff between "cleanliness" and "performance". I agree that it would be cleaner to do the installation in the Jenkinsfile, but then we have to run `swupd bundle-add` every single time that a Jenkins job runs. The way it is now, the build image gets cached in the private Docker registry.

Having said that, at least on my desktop machine `swupd bundle-add sudo openssh-client` is fast (just a few seconds). But I'd prefer to keep it here nonetheless because performance and reliability of remote network access is a bit unpredictable - it might be slower in the CI.

Reply:
> I agree that it would be cleaner to do the installation in the Jenkinsfile, but then we have to run `swupd bundle-add` every single time that a Jenkins job runs.

Does installing them in the `Create build environment` stage helps? As this change altering the build image anyway.

Reply:
Hmm, a bit. We still run "swupd bundle-add", but the result should end up in the cached image. Okay, let's move it.

Reply:
Done.